### PR TITLE
Add content disposition to view, download or both

### PR DESCRIPTION
**Pull request by andresilvagomez**

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -11,5 +11,5 @@
 
 Route::get('/files', 'FilesController@index');
 Route::post('/files', 'FilesController@store');
-Route::get('/files/{fileKey}', 'FilesController@download')->where('fileKey', '.*');
+Route::get('/files/{fileKey}', 'FilesController@file')->where('fileKey', '.*');
```
Comment (ahmedkandel) [inline routes/api.php]: I don't think we need to change the method name, As _inline_ & _attachment_ are actually downloading the file to the browser.

File: src/NovaS3MultipartUpload.php [hunk 2]
```diff
@@ -506,6 +507,17 @@ public function fileManagerSelectionType($string)
         return $this->withMeta(['fileManagerSelectionType' => $string]);
     }
 
+    /**
+     * The content disposition for the files.
+     *
+     * @param  string  $type  (‘view’, ‘download’, ‘both’)
```
Comment (ahmedkandel) [inline src/NovaS3MultipartUpload.php]: Better use array e.g. ['inline'] or ['attachement'] or ['inline', 'attachement'] then pass the value directly to `withMeta`

File: src/Http/Controllers/FilesController.php [hunk 3]
```diff
@@ -177,21 +177,25 @@ private function prepareDataForInsertion($request)
...
 
         abort_unless($file && Storage::disk($this->tool->disk)->exists($file['fileKey']), 404);
 
+        $disposition = $contentDisposition === 'view' ? [] : ['ResponseContentDisposition' => 'attachment; filename="' . ($file['fileName'] ?? basename($file['fileKey'])) . '"'];
+
         return [
             'temporaryUrl' => Storage::disk($this->tool->disk)->temporaryUrl(
                 $file['fileKey'],
                 now()->addMinutes(5),
-                ['ResponseContentDisposition' => 'attachment; filename="' . ($file['fileName'] ?? basename($file['fileKey'])) . '"'],
```
Comment (ahmedkandel) [inline src/Http/Controllers/FilesController.php:194]: you can simplify as `['ResponseContentDisposition' => $request->query('contentDisposition', 'attachment') . '; filename="' . ($file['fileName'] ?? basename($file['fileKey'])) . '"'],`

File: resources/js/components/FileCard.vue [hunk 4]
```diff
@@ -26,7 +26,18 @@
         >
             <button
                 type="button"
-                v-if="withMeta.canDownload"
+                @keydown.enter.prevent="viewFile"
+                @click.prevent="viewFile"
+                v-tooltip.click="__('View')"
+                v-if="withMeta.canDownload && withMeta.contentDisposition !== 'download'"
```
Comment (ahmedkandel) [inline resources/js/components/FileCard.vue]: As contentDisposition will be an array/object use this `withMeta.contentDisposition.includes('attachement')`

File: resources/js/components/FileCard.vue [hunk 5]
```diff
@@ -110,10 +121,26 @@ export default {
 
     methods:
     {
+        viewFile()
```
Comment (ahmedkandel) [inline resources/js/components/FileCard.vue]: No need for a new function you can pass the contentDisposition as a paramter to `downloadFile()` function and then to the `Nova.request().get()`

File: resources/js/components/FileCard.vue [hunk 6]
```diff
@@ -26,7 +26,18 @@
...
             <button
                 type="button"
-                v-if="withMeta.canDownload"
+                @keydown.enter.prevent="viewFile"
+                @click.prevent="viewFile"
+                v-tooltip.click="__('View')"
+                v-if="withMeta.canDownload && withMeta.contentDisposition !== 'download'"
+                class="cursor-pointer dim btn btn-link text-primary inline-flex items-center ml-3"
+            >
+                <icon type="view" view-box="0 0 24 24" width="16" height="16" />
```
Comment (ahmedkandel) [inline resources/js/components/FileCard.vue]: Better use this icon https://github.com/tailwindlabs/heroicons/blob/master/optimized/outline/external-link.svg and keep the eye/view icon for viewing the model resource detail page

**Review by ahmedkandel — CHANGES_REQUESTED** [review]
Thanks, @andresilvagomez for the PR please have a look at my review.

File: resources/js/components/FileCard.vue (hunk 6 above)
Comment (andresilvagomez) [reply resources/js/components/FileCard.vue]: I'm not sure how put this icon,

I used svg

Comment (ahmedkandel) [reply resources/js/components/FileCard.vue]: Try this
```
<svg
    xmlns="http://www.w3.org/2000/svg"
    class="fill-current"
    width="16"
    height="16"
    viewBox="0 0 24 24"
    aria-labelledby="link"
    role="presentation"
  >
    <path d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"/>
</svg>
```

Comment (andresilvagomez) [reply resources/js/components/FileCard.vue]: I used it !!!

**Pull request merged**